src/main/features/dashboard/claims-state-machine/full-defence-transitions.ts

```diff
@@ -0,0 +1,112 @@
+
+import * as StateMachine from '@taoqf/javascript-state-machine'
+import { Claim } from 'claims/models/claim'
+
+import { FreeMediationOption } from 'forms/models/freeMediation'
+import { DefenceType } from 'claims/models/response/defenceType'
+import { ClaimantResponseType } from 'claims/models/claimant-response/claimantResponseType'
+import * as _ from 'lodash'
+import * as path from 'path'
+
+export function FullDefenceTransitions (claim: Claim) {
+  return new StateMachine({
+    init : 'full-defence',
+    transitions : [
+
+      { name : 'checkAlreadyPaid', from: 'full-defence', to: 'fd-already-paid' },
+      { name : 'checkAlreadyPaidResponse', from: ['full-defence','fd-already-paid'], to: 'fd-already-paid-response' },
+      { name : 'checkAlreadyPaidAccept', from: ['full-defence','fd-already-paid-response'], to: 'fd-already-paid-accept' },
+      { name : 'checkAlreadyPaidReject', from: ['full-defence','fd-already-paid-response'], to: 'fd-already-paid-reject' },
+
+      { name : 'checkRejectWithMediation', from: 'full-defence', to: 'fd-reject-with-mediation' },
+      { name : 'checkRejectWithoutMediation', from: 'full-defence', to: 'fd-reject-without-mediation' },
+
+      { name : 'checkSettlementOfferWithMediation', from: ['full-defence', 'fd-reject-with-mediation'], to: 'fd-settlement-offer-with-mediation' },
+      { name : 'checkSettlementOfferWithoutMediation', from: ['full-defence', 'fd-reject-without-mediation'], to: 'fd-settlement-offer-without-mediation' },
+
+      { name : 'checkSettlementOfferRejectWithMediation', from: ['full-defence', 'fd-settlement-offer-with-mediation'], to: 'fd-settlement-offer-reject-with-mediation' },
+      { name : 'checkSettlementOfferRejectWithoutMediation', from: ['full-defence', 'fd-settlement-offer-without-mediation'], to: 'fd-settlement-offer-reject-without-mediation' },
+
+      { name : 'checkSettlementOfferAcceptWithMediation', from: ['full-defence', 'fd-settlement-offer-with-mediation'], to: 'fd-made-agreement-with-mediation' },
+      { name : 'checkSettlementOfferAcceptWithoutMediation', from: ['full-defence', 'fd-settlement-offer-without-mediation'], to: 'fd-made-agreement-without-mediation' },
+
+      { name : 'checkSettledByAgreement', from: ['full-defence', 'fd-settlement-offer-with-mediation','fd-settlement-offer-without-mediation', 'fd-made-agreement-with-mediation','fd-made-agreement-without-mediation'], to: 'fd-settled-with-agreement' },
+      { name : 'checkSettledByAgreement', from: ['full-defence', 'fd-settlement-offer-reject-without-mediation'], to: 'fd-settled' }
+    ],
+    methods: {
+
+      onInvalidTransition (transition: string, from: string, to: string): void {
+        // When the transaction is not allowed then state is going to remain same
```

Comment:
This is good. Just out of interest, did you see this happen anywhere? what should we do if it happens? how are we planning to handle it?